pandas/core/dtypes/dtypes.py

```diff
@@ -361,11 +359,16 @@ def _update_dtype(self, dtype):
                    'got {dtype!r}').format(dtype=dtype)
             raise ValueError(msg)
 
```

Comment:
separate, I don't think we need to have private methods on CDT. e.g. ``validate_ordered`` and ``update_dtype`` (maybe more). As this is *used* in other contexts. so a de-privatize PR should be great.

Reply:
actually can you change this?

Reply:
done for `_update_dtype`, `_validate_ordered`, and `_validate_categories`